src/set_theory/surreal.lean

```diff
@@ -173,15 +173,134 @@ theorem mul_zero_equiv (x : pgame) : (x * 0).equiv 0 :=
 /-- `0 * x` has exactly the same moves as `0`. -/
 def zero_mul_relabelling : Π (x : pgame), relabelling (0 * x) 0
 | (mk xl xr xL xR) :=
-⟨by fsplit; rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
- by fsplit; rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
- by rintro (⟨⟨⟩,_⟩ | ⟨⟨⟩,_⟩),
+⟨by fsplit; rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
+ by fsplit; rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
+ by rintro (⟨⟨⟩,_⟩|⟨⟨⟩,_⟩),
  by rintro ⟨⟩⟩
 
 /-- `0 * x` is equivalent to `0`. -/
 theorem zero_mul_equiv (x : pgame) : (0 * x).equiv 0 :=
 (zero_mul_relabelling x).equiv
 
+/-- Local tactic useful for resolving goals involving equivalences between types,
+such as, a × (b ⊕ c) ≃ a × b ⊕ a × c. -/
+meta def try_inl_inr : tactic unit :=
+`[assumption]
+    <|> (do `[apply sum.inl], try_inl_inr )
+    <|> (do `[apply sum.inr], try_inl_inr )
+    <|> (do `[apply prod.mk], try_inl_inr, try_inl_inr )
```

Comment:
Could you try out `solve_by_elim [sum.inl, sum.inr, prod.mk]`? You may have to increase the max depth, as `solve_by_elim` has a very low default (3, I think).

Reply:
Worked at depth 5 :D 
Thanks, I'll make the change.